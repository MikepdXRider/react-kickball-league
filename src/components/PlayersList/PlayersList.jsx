import React from 'react';
import PropTypes from 'prop-types';
import PlayerCard from '../PlayerCard/PlayerCard.jsx';
import { Link } from 'react-router-dom';


export default function PlayersList({playersDataArr}) {
    return (
        <ul>
            {
                playersDataArr.map(playerDataObj => {
                    return (
                        <Link to={`/players/${playerDataObj.id}`}>
                            <li key={playerDataObj.id}>
                                <PlayerCard playerDataObj={playerDataObj} />
                            </li>
                        </Link>
                    )
                })
            }
        </ul>
    )
};

PlayersList.propTypes = {
    playersDataArr: PropTypes.arrayOf(PropTypes.object)
};