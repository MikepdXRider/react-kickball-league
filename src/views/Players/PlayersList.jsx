import React from 'react';
import { Link } from 'react-router-dom';
import { useEffect, useState } from 'react/cjs/react.development';
import PlayerCard from '../../components/PlayerCard/PlayerCard.jsx';
import { getPlayers } from '../../services/players.jsx';

export default function PlayersList() {
    
    const [playersDataArr, setPlayersDataArr] = useState([]);
    const [isLoading, setIsLoading] = useState(false);

    useEffect(() => {
        setIsLoading(true);
        async function getPlayersData() {
            const newPlayersDataArr = await getPlayers();
            // console.log('getPlayers fn returns: ', newPlayersDataArr);
            setPlayersDataArr(newPlayersDataArr);
        }

        getPlayersData();
        setIsLoading(false);
    }, []);
   
    return (
        <main>
            {
                isLoading
                ? <h1>Loading...</h1> 
                : <section>
                    <ul>
                        {playersDataArr.map(playerDataObj => {
                            return (
                                <li key={playerDataObj.id}>
                                    <Link to={`/players/${playerDataObj.id}`}>
                                        <PlayerCard playerDataObj={playerDataObj}/>
                                    </Link>
                                </li>
                            )
                        })}
                    </ul>
                </section>
            }
        </main>
    )
}
