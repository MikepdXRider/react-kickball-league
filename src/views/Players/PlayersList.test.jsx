import { screen } from "@testing-library/react";
import { render } from "@testing-library/react";
import { MemoryRouter } from "react-router";
import PlayersList from "./PlayersList.jsx";


it ('renders PlayersList page', async () => {
    const { container } = render(
            <MemoryRouter>
                <PlayersList />
            </MemoryRouter>
        )
    // count nest line 14 in waitFor() - maybe. 
    const playerName = await screen.findByText(/Add Player/i);

    expect(playerName).toBeInTheDocument();
    expect(container).toMatchSnapshot();
})