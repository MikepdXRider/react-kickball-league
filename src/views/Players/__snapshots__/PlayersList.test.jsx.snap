// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders PlayersList page 1`] = `
<div>
  <main>
    <section>
      <ul>
        <li>
          <a
            href="/players/1"
          >
            <article>
              <h3>
                Ben E. Jetts
              </h3>
              <h6>
                2021-11-22T20:11:15+00:00
              </h6>
              <h5>
                Position: 
                Pitcher
              </h5>
            </article>
          </a>
        </li>
        <li>
          <a
            href="/players/12"
          >
            <article>
              <h3>
                Betty Grey
              </h3>
              <h6>
                2021-11-28T04:55:57+00:00
              </h6>
              <h5>
                Position: 
                Shortstop
              </h5>
            </article>
          </a>
        </li>
        <li>
          <a
            href="/players/7"
          >
            <article>
              <h3>
                Bud E. Guy
              </h3>
              <h6>
                2021-11-28T04:48:43+00:00
              </h6>
              <h5>
                Position: 
                Outfield
              </h5>
            </article>
          </a>
        </li>
        <li>
          <a
            href="/players/8"
          >
            <article>
              <h3>
                Buster Rimes
              </h3>
              <h6>
                2021-11-28T04:50:00+00:00
              </h6>
              <h5>
                Position: 
                Shortstop
              </h5>
            </article>
          </a>
        </li>
        <li>
          <a
            href="/players/9"
          >
            <article>
              <h3>
                Deb E. Downer
              </h3>
              <h6>
                2021-11-28T04:50:18+00:00
              </h6>
              <h5>
                Position: 
                Pitcher
              </h5>
            </article>
          </a>
        </li>
        <li>
          <a
            href="/players/13"
          >
            <article>
              <h3>
                Dwayne Elizondo Mountain Dew Herbert Camacho
              </h3>
              <h6>
                2021-11-28T04:56:31+00:00
              </h6>
              <h5>
                Position: 
                Pitcher
              </h5>
            </article>
          </a>
        </li>
        <li>
          <a
            href="/players/15"
          >
            <article>
              <h3>
                Frito Pendejo
              </h3>
              <h6>
                2021-12-01T05:08:58+00:00
              </h6>
              <h5>
                Position: 
                Outfield
              </h5>
            </article>
          </a>
        </li>
        <li>
          <a
            href="/players/5"
          >
            <article>
              <h3>
                Gary Coolman
              </h3>
              <h6>
                2021-11-28T04:47:26+00:00
              </h6>
              <h5>
                Position: 
                Pitcher
              </h5>
            </article>
          </a>
        </li>
        <li>
          <a
            href="/players/2"
          >
            <article>
              <h3>
                Hans Olo
              </h3>
              <h6>
                2021-11-22T20:12:14+00:00
              </h6>
              <h5>
                Position: 
                Catcher
              </h5>
            </article>
          </a>
        </li>
        <li>
          <a
            href="/players/10"
          >
            <article>
              <h3>
                Jayne Phonda
              </h3>
              <h6>
                2021-11-28T04:53:23+00:00
              </h6>
              <h5>
                Position: 
                Catcher
              </h5>
            </article>
          </a>
        </li>
        <li>
          <a
            href="/players/14"
          >
            <article>
              <h3>
                Joe Bauers
              </h3>
              <h6>
                2021-12-01T05:07:28+00:00
              </h6>
              <h5>
                Position: 
                Catcher
              </h5>
            </article>
          </a>
        </li>
        <li>
          <a
            href="/players/11"
          >
            <article>
              <h3>
                Kara Whollot
              </h3>
              <h6>
                2021-11-28T04:54:03+00:00
              </h6>
              <h5>
                Position: 
                Outfield
              </h5>
            </article>
          </a>
        </li>
        <li>
          <a
            href="/players/3"
          >
            <article>
              <h3>
                Ron Long
              </h3>
              <h6>
                2021-11-22T20:13:14+00:00
              </h6>
              <h5>
                Position: 
                Outfield
              </h5>
            </article>
          </a>
        </li>
        <li>
          <a
            href="/players/4"
          >
            <article>
              <h3>
                Shorty Biggum
              </h3>
              <h6>
                2021-11-22T20:15:04+00:00
              </h6>
              <h5>
                Position: 
                Shortstop
              </h5>
            </article>
          </a>
        </li>
        <li>
          <a
            href="/players/6"
          >
            <article>
              <h3>
                Terry Cruise
              </h3>
              <h6>
                2021-11-28T04:47:40+00:00
              </h6>
              <h5>
                Position: 
                Catcher
              </h5>
            </article>
          </a>
        </li>
      </ul>
    </section>
  </main>
</div>
`;
